File: problem-solving/_lc-674-easy-longest-continuous-increasing-subsequence/1-brute-force.go
/*
https://leetcode.com/problems/longest-continuous-increasing-subsequence/

Complexity
----------
Time: O(n^2)
Space: O(1)
*/

func findLengthOfLCIS(nums []int) int {
    lcis := 1
    n := len(nums)
    for i := 0; i < n; i++ {
        cis := 1
        for j := i + 1; j < n; j++ {
            if nums[j - 1] >= nums[j] {
                break
            }
            cis += 1
        }
        if cis > lcis {
            lcis = cis
        }
    }
    return lcis
}

File: problem-solving/lc-674-easy-longest-continuous-increasing-subsequence/1-brute-force.py
"""
https://leetcode.com/problems/longest-continuous-increasing-subsequence/description/

Complexity
----------
Time: O(n^2)
Space: O(1)
"""

class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        ans = 1
        n = len(nums)
        for i in range(n):
            v = 1
            for j in range(i + 1, n):
                if nums[j - 1] >= nums[j]:
                    break
                v += 1
            ans = max(ans, v)
        return ans
